dessia_common/utils/diff.py

```diff
@@ -38,11 +38,19 @@ def diff(value1, value2, path='#'):
     if isinstance(value1, dict):
         return dict_diff(value1, value2, path=path)
     # elif hasattr(value1, '_data_eq'):
-
-    # Should be object
-    if hasattr(value1, '_data_eq'):
-        # DessiaObject
-        if value1._data_eq(value2):
+    else:
+        # Should be object
+        # print(value1, isinstance(value1, dc.DessiaObject),  hasattr(value1, '_data_eq'))
+        # if isinstance(value1, dc.DessiaObject):
+        if hasattr(value1, '_data_eq'):
+            # DessiaObject
+            if value1._data_eq(value2):
+                return [], [], []
+        elif hasattr(value2, '_data_eq'):
```

Comment:
This case looks weird. If value 2 have _data_eq that value1 do not, hence they are necessarily different, right ? It looks like we should never end up in here